File: src/src/org/github/olloginov/ideataskbrowser/TaskBrowser.java
package org.github.olloginov.ideataskbrowser;

import com.intellij.openapi.components.PersistentStateComponent;
import org.github.olloginov.ideataskbrowser.config.TaskBrowserConfig;

import javax.swing.tree.TreeModel;

public interface TaskBrowser extends TaskBrowserServiceState, PersistentStateComponent<TaskBrowserConfig> {
    void refresh();

    void reloadChanges();

    TreeModel getFilteredModel();
}

File: src/main/kotlin/org/github/olloginov/ideataskbrowser/TaskBrowser.kt
package org.github.olloginov.ideataskbrowser;

import com.intellij.openapi.components.PersistentStateComponent;
import org.github.olloginov.ideataskbrowser.config.TaskBrowserConfig;

import javax.swing.tree.TreeModel;

interface TaskBrowser : TaskBrowserServiceState, PersistentStateComponent<TaskBrowserConfig> {
	fun refresh()

	fun reloadChanges()

	fun getFilteredModel(): TreeModel
}
